### PR TITLE
EOS-16179:[DTM0]

### PR DESCRIPTION
**Pull request by mehjoshi**
Added dtm0 fop request, reply structures, operation and message types,
markers in code where dtm0 operation related credit calculation should
be performed and dtm0 log routines should be called for the basic case
of executing a dtm0 request.
Also added DLD for dtm0 service.

Signed-off-by: Mehul Manharlal Joshi <mehul.joshi@seagate.com>

### CODE REVIEW

File: dtm0/fop.c [hunk 1]
```diff
@@ -110,46 +115,91 @@ int m0_dtm0_fop_init(void)
...
+static int dtm0_fom_create(struct m0_fop *fop,
+			  struct m0_fom **out, struct m0_reqh *reqh)
+{
+	struct dtm0_fom    *fom;
+	struct m0_fom     *fom0;
+	struct m0_fop     *repfop;
+	struct dtm0_rep_fop *reply;
+
+	if (!dtm0_service_started(fop, reqh))
+		return M0_ERR(-EAGAIN);
```
Comment (just-now) [inline dtm0/fop.c:182]: Do we really need this check?
I feel that we never hit -EAGAIN case here. Also, I think that similar check is somewhere during processing on the fom checking that service instance is proper.

**Review by just-now — COMMENTED** [review]
Looks good.

File: dtm0/fop.c (hunk 1 above)
Comment (mehjoshi) [reply dtm0/fop.c:182]: I think this may not be needed .. will remove it.

**Review by just-now — APPROVED** [review]
I'm OK to merge.

File: cas/client.c [hunk 2]
```diff
@@ -176,6 +176,7 @@ static int creq_op_alloc(uint64_t           recs_nr,
 	} else {
 		op->cg_rec.cr_nr = recs_nr;
 		op->cg_rec.cr_rec = rec;
+		m0_dtm0_tx_desc_init_none(&op->cg_txd);
```
Comment (just-now) [inline cas/client.c:179]: test, ignore this.
cc @just-now

**Pull request merged**